File: backend/model/booked_slot_address.go
package model

import "gorm.io/gorm"

type BookedSlotAddress struct {
	gorm.Model
	Name      string  `gorm:"not null" json:"name"`
	Address   string  `gorm:"not null" json:"address"`
	Latitude  float64 `gorm:"not null" json:"latitude"`
	Longitude float64 `gorm:"not null" json:"longitude"`
	Notes     *string `gorm:"default:null" json:"notes"`
}

type BookedSlotAddressDTO struct {
	ID        uint    `json:"id"`
	Name      string  `json:"name"`
	Address   string  `json:"address"`
	Latitude  float64 `json:"latitude"`
	Longitude float64 `json:"longitude"`
	Notes     *string `json:"notes"`
}

File: frontend/lib/model/booked_slot_address.dart
import 'package:freezed_annotation/freezed_annotation.dart';

part 'booked_slot_address.freezed.dart';

part 'booked_slot_address.g.dart';

@freezed
class BookedSlotAddress with _$BookedSlotAddress {
  factory BookedSlotAddress({
    required int id,
    required String name,
    required String address,
    required double latitude,
    required double longitude,
    required String? notes,
  }) = _BookedSlotAddress;

  factory BookedSlotAddress.fromJson(Map<String, dynamic> json) =>
      _$BookedSlotAddressFromJson(json);
}
